qutebrowser/browser/greasemonkey.py

```diff
@@ -0,0 +1,289 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016-2017 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Load, parse and make avalaible greasemonkey scripts."""
+
+import re
+import os
+import json
+from fnmatch import fnmatch
+from functools import partial
+
+from PyQt5.QtCore import QDir, pyqtSignal, QObject
+
+from qutebrowser.config import config
+from qutebrowser.utils import log, standarddir
+from qutebrowser.commands import cmdutils
+
+
+def _scripts_dir():
+    """Get the directory of the scripts."""
+    directory = config.val.content.javascript.userjs_directory
+    if directory is None:
+        directory = os.path.join(standarddir.data(), 'greasemonkey')
+    return directory
+
+
+class GreasemonkeyScript:
+    """Container class for userscripts, parses metadata blocks."""
+
+    GM_BOOTSTRAP_TEMPLATE = r"""var _qute_script_id = "__gm_{scriptName}";
```

Comment:
Like outlined in https://github.com/qutebrowser/qutebrowser/pull/664#issuecomment-100826271 I think it'd be nicer to have this in its own `greasemonkey.js` file and use jinja to interpolate it, then you can use `{` and `}` normally as well.